src/lib/metricsMgr/MetricsManager.h

```diff
@@ -41,28 +94,33 @@
 *    initial map created at constructor time)
 * 3. Destroy method, releasing all the maps in cascade (probably never used, as the
 *    singleton object in CB using this class will be destroyed at the end, but do it
-*    for class compleness)
+*    for class completeness)
 * 4. reset() method implementation (not delete maps, only set metrics to 0)
-* 5. toJson() to be splitted into 3 methods (2 of them private)
+* 5. toJson() to be split into 3 methods (2 of them private)
 * 6. Multi-thread safeness. Probably the same sem-based strategy used in AlarmManager
 *    could be used.
 * 7. Use 'long long' instead of 'int'
-* 8. (Usure) We could need maps for metrics different for int. In that case, implement
-*    it (and the accum method) using templates, to avoid repeat the same implementation
+* 8. (Unsure) We could need maps for metrics different for int. In that case, implement
+*    it (and the add method) using templates, to avoid repeating the same implementation
 *    N times
 */
 class MetricsManager
 {
  private:
-  std::map<std::string, std::map<std::string, std::map<std::string, int>*>* >  metrics;
+  std::map<std::string, std::map<std::string, std::map<std::string, int>*>*>  metrics;
+  sem_t           sem;
```

Comment:
Is this sem included in the https://fiware-orion.readthedocs.io/en/master/admin/statistics/index.html#semwait-block statistics??

I think it would be a good idea, in order to see in the future how the metrics subsystem is impacting in performance.